packages/tiger-san/src/form/input.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+
+const getInputClassName = ({ type }) => (
+  classnames('input', {
+    'input--select': type === 'select',
+    'input--textarea': type === 'textarea'
+  })
+)
+
+const getInputComponent = ({ type }) => {
+  return ['select', 'textarea'].includes(type) ? type : 'input'
+}
+
+export const Input = React.forwardRef(({
```

Comment:
what about other props like `onClick`, `onChange`, `disabled` ?

Reply:
one other question, I'm guessing you'll be adding components like `InputSelect` and `InputTextArea` that returns `Input` with specific props. To handle that hierarchy, `Input` would need `{...props}` to map html element specific props.
e.g. `select` element has an attribute `multiple` and needs to be mapped in`Input`
e.g. `textarea` element has an attribute `rows` and needs to be mapped in `Input`

Reply:
Good question ... are you suggesting that we talk the approach:

```js
import { InputSelect, InputText, InputTextArea } from '@lemonpeach/tiger-san'

<Form>
  <InputSelect ... />
  <InputText ... />
```

I know this is how we have usually done this ... but I was taking a different approach here -- there would only be one `Input` and you would pass a `type` attribute` to it. `Input` component would handle the rest.

Internally we could have `InputSelect` handle that `type`, but I was thinking of maybe changing that too ... we could have some kind of "mixin" behaviour.

Thoughts?

Reply:
oh I see, I'm cool the new approach as long as `Input` doesn't get too verbose for handing various `types`!